vispy/scene/cameras/panzoom.py

```diff
@@ -273,36 +275,32 @@ def _update_transform(self):
         # Apply mapping between viewbox and cam view
         self.transform.set_mapping(self._real_rect, vbr, update=False)
         # Scale z, so that the clipping planes are between -alot and +alot
-        self.transform.zoom((1, 1, 1/d))
+        self.transform.zoom((1, 1, 1 / d))
 
         # We've now set self.transform, which represents our 2D
         # transform When up is +z this is all. In other cases,
         # self.transform is now set up correctly to allow pan/zoom, but
         # for the scene we need a different (3D) mapping. When there
         # is a minus in up, we simply look at the scene from the other
         # side (as if z was flipped).
-
-        if self.up == '+z':
-            thetransform = self.transform
-        else:
-            rr = self._real_rect
-            tr = MatrixTransform()
-            d = d if (self.up[0] == '+') else -d
-            pp1 = [(vbr.left, vbr.bottom, 0), (vbr.left, vbr.top, 0),
-                   (vbr.right, vbr.bottom, 0), (vbr.left, vbr.bottom, 1)]
-            # Get Mapping
-            if self.up[1] == 'z':
-                pp2 = [(rr.left, rr.bottom, 0), (rr.left, rr.top, 0),
-                       (rr.right, rr.bottom, 0), (rr.left, rr.bottom, d)]
-            elif self.up[1] == 'y':
-                pp2 = [(rr.left, 0, rr.bottom), (rr.left, 0, rr.top),
-                       (rr.right, 0, rr.bottom), (rr.left, d, rr.bottom)]
-            elif self.up[1] == 'x':
-                pp2 = [(0, rr.left, rr.bottom), (0, rr.left, rr.top),
-                       (0, rr.right, rr.bottom), (d, rr.left, rr.bottom)]
-            # Apply
-            tr.set_mapping(np.array(pp2), np.array(pp1))
-            thetransform = tr
+        rr = self._real_rect
+        d = d if (self.up[0] == '+') else -d
+        pp1 = [(vbr.left, vbr.bottom, 0), (vbr.left, vbr.top, 0),
+               (vbr.right, vbr.bottom, 0), (vbr.left, vbr.bottom, 1)]
+        # Get Mapping
+        if self.up[1] == 'z':
+            pp2 = [(rr.left, rr.bottom, 0), (rr.left, rr.top, 0),
+                   (rr.right, rr.bottom, 0), (rr.left, rr.bottom, d)]
+        elif self.up[1] == 'y':
+            pp2 = [(rr.left, 0, rr.bottom), (rr.left, 0, rr.top),
+                   (rr.right, 0, rr.bottom), (rr.left, d, rr.bottom)]
+        elif self.up[1] == 'x':
+            pp2 = [(0, rr.left, rr.bottom), (0, rr.left, rr.top),
+                   (0, rr.right, rr.bottom), (d, rr.left, rr.bottom)]
+
+        # Apply
+        self.tf_mat.reset()
```

Comment:
Is this needed? From what I can tell in the `MatrixTransform` sets `self.matrix` in `set_mapping` which then updates the shaders. This (the reset followed by the `set_mapping`) is actually causing to calls to `update` which is doing double the number of draws (it sends a "changed" event out to the vispy system).

Reply:
Good call